public/pages/workflow_detail/utils/workflow_to_template_utils.ts

```diff
@@ -100,49 +165,54 @@ function processorConfigsToTemplateNodes(
     }
   });
 
-  const ingestPipelineName = generateId('ingest_pipeline');
-  return [
-    {
-      id: ingestPipelineName,
-      type: WORKFLOW_STEP_TYPE.CREATE_INGEST_PIPELINE_STEP_TYPE,
-      user_inputs: {
-        pipeline_id: ingestPipelineName,
-        configurations: {
-          description: 'An ingest pipeline',
-          processors: processorsList,
-        },
-      },
-    } as CreateIngestPipelineNode,
-  ];
+  return processorsList;
 }
 
 // General fn to convert an index config to a final CreateIndexNode template node.
-// Requires any ingest/pipeline node details to set any defaults
+// Requires any ingest/pipeline node details to set any defaults, if applicable.
 function indexConfigToTemplateNode(
   indexConfig: IndexConfig,
-  ingestPipelineNode: CreateIngestPipelineNode
+  ingestPipelineNode?: CreateIngestPipelineNode,
+  searchPipelineNode?: CreateSearchPipelineNode
 ): CreateIndexNode {
   const indexName = indexConfig.name.value as string;
 
   // TODO: extract model details to determine the mappings
-
-  // index mappings are different per use case
   const finalIndexMappings = {
     properties: {},
   } as IndexMappings;
 
+  let finalPreviousNodeInputs = {};
+  let finalSettings = {};
+  if (ingestPipelineNode) {
+    finalPreviousNodeInputs = {
+      ...finalPreviousNodeInputs,
+      [ingestPipelineNode.id]: 'pipeline_id',
+    };
+    finalSettings = {
+      ...finalSettings,
+      default_pipeline: `\${{${ingestPipelineNode.id}.pipeline_id}}`,
+    };
+  }
+  if (searchPipelineNode) {
+    finalPreviousNodeInputs = {
+      ...finalPreviousNodeInputs,
+      [searchPipelineNode.id]: 'pipeline_id',
+    };
+    finalSettings = {
+      ...finalSettings,
+      ['index.search.default_pipeline']: `\${{${searchPipelineNode.id}.pipeline_id}}`,
+    };
+  }
```

Comment:
To avoid duplication. Something like:
```suggestion
  const updateFinalInputsAndSettings = (node) => {
  if (node) {
    finalPreviousNodeInputs = {
      ...finalPreviousNodeInputs,
      [node.id]: 'pipeline_id',
    };

    const pipelineKey = node.type === WORKFLOW_STEP_TYPE.CREATE_INGEST_PIPELINE_STEP_TYPE
      ? 'default_pipeline'
      : 'index.search.default_pipeline';

    finalSettings = {
      ...finalSettings,
      [pipelineKey]: `\${{${node.id}.pipeline_id}}`,
    };
  }
};

updateFinalInputsAndSettings(ingestPipelineNode);
updateFinalInputsAndSettings(searchPipelineNode);
```

Reply:
Sure, thanks for the suggestion! will update.